subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -953,23 +915,77 @@ static void pe_chunk_received_run(void *obj)
 	}
 }
 
-/**
- * @brief Suspend State
+/*
+ * @brief Super State for any message that requires
+ *	  Sender Response Timer functionality
  */
-static void pe_suspend_entry(void *obj)
+static void pe_sender_response_run(void *obj)
```

Comment:
What's the main advantage you see of making this a parent state as opposed to a set of helpers (as we have in our EC codebase)?

Reply:
This approach seems cleaner to me and it utilizes the hierarchical functionality of the stack.

Reply:
I know we discussed this offline but I still think my current approach is easier to read and maintain. Also, one point for using a hierarchical state machine is so we don't have to duplicate behavior across many states.